internal/k8s/client.go

```diff
@@ -192,9 +192,9 @@ func (w *ClientWrapper) CoreDNSMatch() (bool, error) {
 			continue
 		}
 
-		split := strings.Split(c.Image, ":")
-		if len(split) == 2 {
-			version = split[1]
+		split := strings.LastIndex(c.Image, ":")
+		if split != -1 {
+			version = c.Image[(split + 1):]
```

Comment:
This works, but it's not nearly as elegant as @jlevesy's suggestion, which doesn't include an `if` statement and is overall much simpler:

```go
	sp := strings.Split(c.Image, ":")
	version = sp[len(sp)-1]
```